src/main/java/it/gov/pagopa/pu/debtpositions/service/statusalign/DebtPositionHierarchyStatusAlignerServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package it.gov.pagopa.pu.debtpositions.service.statusalign;
+
+import it.gov.pagopa.pu.debtpositions.dto.generated.InstallmentStatus;
+import it.gov.pagopa.pu.debtpositions.dto.generated.IudSyncStatusUpdateDTO;
+import it.gov.pagopa.pu.debtpositions.exception.custom.InvalidStatusException;
+import it.gov.pagopa.pu.debtpositions.model.DebtPosition;
+import it.gov.pagopa.pu.debtpositions.repository.DebtPositionRepository;
+import it.gov.pagopa.pu.debtpositions.repository.InstallmentNoPIIRepository;
+import it.gov.pagopa.pu.debtpositions.service.statusalign.debtposition.DebtPositionInnerStatusAlignerService;
+import it.gov.pagopa.pu.debtpositions.service.statusalign.paymentoption.PaymentOptionInnerStatusAlignerService;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+import static it.gov.pagopa.pu.debtpositions.dto.generated.InstallmentStatus.TO_SYNC;
+
+@Service
+public class DebtPositionHierarchyStatusAlignerServiceImpl implements DebtPositionHierarchyStatusAlignerService {
+
+  private final DebtPositionRepository debtPositionRepository;
+  private final InstallmentNoPIIRepository installmentNoPIIRepository;
+  private final PaymentOptionInnerStatusAlignerService paymentOptionInnerStatusAlignerService;
+  private final DebtPositionInnerStatusAlignerService debtPositionInnerStatusAlignerService;
+
+
+  public DebtPositionHierarchyStatusAlignerServiceImpl(DebtPositionRepository debtPositionRepository,
+                                                       InstallmentNoPIIRepository installmentNoPIIRepository, PaymentOptionInnerStatusAlignerService paymentOptionInnerStatusAlignerService, DebtPositionInnerStatusAlignerService debtPositionInnerStatusAlignerService) {
+    this.debtPositionRepository = debtPositionRepository;
+    this.installmentNoPIIRepository = installmentNoPIIRepository;
+    this.paymentOptionInnerStatusAlignerService = paymentOptionInnerStatusAlignerService;
+    this.debtPositionInnerStatusAlignerService = debtPositionInnerStatusAlignerService;
+  }
+
+  @Transactional
+  @Override
+  public void finalizeSyncStatus(Long debtPositionId, Map<String, IudSyncStatusUpdateDTO> syncStatusDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findOneWithAllDataByDebtPositionId(debtPositionId);
+
+    debtPosition.getPaymentOptions().forEach(paymentOption ->
+      paymentOption.getInstallments().stream()
+        .filter(installment -> TO_SYNC.equals(installment.getStatus()))
+        .filter(installment -> syncStatusDTO.containsKey(installment.getIud()))
```

Comment:
can you put just 1 filter operation in order to:
* log as ERROR a TO_SYNC installment not present in the map
* log as ERROR an installment in the map, but without TO_SYNC status
The filter will return the AND between the actually configured conditions